common/persistence/pinotResponseComparator.go

```diff
@@ -0,0 +1,487 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package persistence
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/uber/cadence/common/log/tag"
+
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/types"
+)
+
+func interfaceToMap(in interface{}) map[string][]byte {
+	if in == nil || in == "" {
+		return map[string][]byte{}
+	}
+
+	v, ok := in.(map[string][]byte)
+	if !ok {
+		panic(fmt.Sprintf("interface to map error in ES/Pinot comparator: %#v", in))
+	}
+
+	return v
+}
+
+func compareSearchAttributes(esSearchAttribute interface{}, pinotSearchAttribute interface{}) error {
+	esAttr, ok := esSearchAttribute.(*types.SearchAttributes)
+	if !ok {
+		panic("interface is not a SearchAttributes! ")
```

Comment:
we can cay is not a ES search attribute, same below. So we can know which part has the issue

Reply:
done